client/src/main/java/com/alibaba/nacos/client/naming/remote/gprc/NamingGrpcClientProxy.java

```diff
@@ -197,8 +253,32 @@ public void doBatchRegisterService(String serviceName, String groupName, List<In
             throws NacosException {
         BatchInstanceRequest request = new BatchInstanceRequest(namespaceId, serviceName, groupName,
                 NamingRemoteConstants.BATCH_REGISTER_INSTANCE, instances);
-        requestToServer(request, BatchInstanceResponse.class);
-        redoService.instanceRegistered(serviceName, groupName);
+        
+        Span span = NamingTrace.getClientNamingWorkerSpan("doBatchRegisterService");
+        try (Scope ignored = span.makeCurrent()) {
+            
+            requestToServer(request, BatchInstanceResponse.class);
+            redoService.instanceRegistered(serviceName, groupName);
+            
+            if (span.isRecording()) {
```

Comment:
Some attributes can be set before doing actions in order to record more information for exceptions.

Reply:
For some attributes, especially those that are expensive to calculate ( eg. some instance states), should be added if span is recording. https://opentelemetry.io/docs/concepts/instrumentation/libraries/#performance However, some attributes in the Nacos client may not need to be operated in this way (in fact, I have made the version number of the nacos client a must-have attribute for each span). I will discuss it with you in detail.